client/lib/purchases/index.js

```diff
@@ -371,15 +376,21 @@ function canExplicitRenew( purchase ) {
 }
 
 function creditCardExpiresBeforeSubscription( purchase ) {
+	const creditCard = purchase?.payment?.creditCard;
+	const purchaseExpiryDate = moment( purchase.expiryDate );
+
 	return (
 		isPaidWithCreditCard( purchase ) &&
 		hasCreditCardData( purchase ) &&
-		purchase.payment.creditCard.expiryMoment.diff( purchase.expiryMoment, 'months' ) < 0
+		creditCard.expiryDate &&
+		moment( creditCard.expiryDate, 'MM/YY' ).diff( purchaseExpiryDate, 'months' ) < 0
```

Comment:
This code is guarded by `isPaidWithCreditCard` and `hasCreditCardData`. If these checks are successful, it more or less means that `purchase.payment.creditCard.expiryDate` is set and valid. The extra checks are not necessary.

Also, the `purchase.expiryDate` string can be passed directly as argument of `diff`:
```js
moment( card.expiryDate ).diff( purchase.expiryDate, 'months' ) < 0
```
The parameter has type `MomentInput`.

An alternative (and better IMO) way to write the condition is:
```js
moment( creditCard.expiryDate, 'MM/YY' ).isBefore( purchase.expiryDate, 'months' )
```